From e80384786d7b7f895c97f2447d4b91af8eb5f0f4 Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Wed, 2 Sep 2020 09:45:17 +0200
Subject: [PATCH 1/2] Revert "Fix static linking with GCC-10"

This reverts commit 5b58a1ebd89a4f05778441814e81817c82193fa3.

This breaks all static builds earlier to gcc 10 :(
Bad testing on my side.
---
 libc/misc/elf/dl-support.c                 | 2 +-
 libpthread/nptl/sysdeps/generic/libc-tls.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libc/misc/elf/dl-support.c b/libc/misc/elf/dl-support.c
index eaf08f5bb..87cd1bb72 100644
--- a/libc/misc/elf/dl-support.c
+++ b/libc/misc/elf/dl-support.c
@@ -29,7 +29,7 @@ void (*_dl_init_static_tls) (struct link_map *) = &_dl_nothread_init_static_tls;
 
 ElfW(Phdr) *_dl_phdr;
 size_t _dl_phnum;
-extern size_t _dl_pagesize;
+size_t _dl_pagesize;
 
 void internal_function _dl_aux_init (ElfW(auxv_t) *av);
 void internal_function _dl_aux_init (ElfW(auxv_t) *av)
diff --git a/libpthread/nptl/sysdeps/generic/libc-tls.c b/libpthread/nptl/sysdeps/generic/libc-tls.c
index a1d2ef913..d746c9a38 100644
--- a/libpthread/nptl/sysdeps/generic/libc-tls.c
+++ b/libpthread/nptl/sysdeps/generic/libc-tls.c
@@ -72,7 +72,7 @@ struct dtv_slotinfo_list *_dl_tls_dtv_slotinfo_list;
 /* Number of modules in the static TLS block.  */
 size_t _dl_tls_static_nelem;
 /* Size of the static TLS block.  */
-extern size_t _dl_tls_static_size;
+size_t _dl_tls_static_size;
 /* Size actually allocated in the static TLS block.  */
 size_t _dl_tls_static_used;
 /* Alignment requirement of the static TLS block.  */
-- 
2.27.0.rc0

